lib/segment/analytics/worker.rb

```diff
@@ -41,19 +41,17 @@ def run
               @batch << @queue.pop
             end
           end
-
           res = Request.new.post @write_key, @batch
 
           @on_error.call res.status, res.error unless res.status == 200
-
           @lock.synchronize { @batch.clear }
         end
       end
 
       # public: Check whether we have outstanding requests.
       #
       def is_requesting?
-        @lock.synchronize { !@batch.empty? }
+        @lock.synchronize { @batch.present? }
```

Comment:
this relies on activesupport, which is a dependency you usually don't want to have in more generic gems.
We might wanna change this to use `any?`